tokio-rustls/src/client.rs

```diff
@@ -136,6 +156,11 @@ where
 
                 // end
                 this.state = TlsState::Stream;
+
+                if let Some(waker) = this.early_waker.take() {
+                    waker.wake();
+                }
```

Comment:
It looks like there's quite a bit of code duplication between the early data handling in `poll_write()` and `poll_flush()` maybe pull that out into a separate method? (Ideally in a separate commit.)

Reply:
agree, we can complete it in a later PR.